crates/sui/src/upgrade_compatibility.rs

```diff
@@ -813,6 +821,43 @@ fn module_compatibility_error_diag(
     Ok(diags)
 }
 
+fn missing_module_diag(
+    module_name: &Identifier,
+    package_path: &PathBuf,
+) -> Result<Diagnostics, Error> {
+    const PACKAGE_TABLE: &str = "[package]";
+    let mut diags = Diagnostics::new();
+
+    // read Move.toml to get the hash and first line start and end
+    let toml_path = package_path.join("Move.toml");
+    let toml_str = fs::read_to_string(&toml_path).context("Unable to read Move.toml")?;
```

Comment:
Didn't we parse the contents already for the FilesSourceText?

Reply:
fixed